### ISSUE TITLE
error connectycube flutter callkit not working in the background

### ISSUE THREAD

**Issue by devFan1** [ConnectyCube/connectycube-flutter-call-kit#96]
error :`FlutterConnectycubeBackgroundExecutor(13254): No one background handler has been registered.`

when the app is in the background the plugin stop working it's like the method  `onCallAcceptedWhenTerminated` is not readble

have anyone faced this issue and solved it please answer

**Comment by TatankaConCube** [comment]
are you sure that your callbacks `onCallRejectedWhenTerminated ` and `onCallAcceptedWhenTerminated` satisfy the main requirements from the [README](https://github.com/ConnectyCube/connectycube-flutter-call-kit#listen-in-the-background-or-terminated-state-android-only)?

**Comment by berger89** [comment]
We have the same problem with `onCallRejectedWhenTerminated` and `onCallAcceptedWhenTerminated` not being called when the app is terminated:
When the app is started in debug mode, these two top-level functions are called.
But when the app is built, the functions are not started.

```
@pragma('vm:entry-point')
Future<dynamic> onCallRejectedWhenTerminated(CallEvent callEvent) async {
  debugPrint("CallEventModel: onCallRejectedWhenTerminated $callEvent");
  ...
}

@pragma('vm:entry-point')
Future<dynamic> onCallAcceptedWhenTerminated(CallEvent callEvent) async {
  debugPrint("CallEventModel: onCallAcceptedWhenTerminated $callEvent");
  ... 
}

class ConnectyCubeEventModel extends ChangeNotifier {

  ...

    ConnectycubeFlutterCallKit.instance.init(
        onCallAccepted: _onCallAccepted,
        onCallRejected: _onCallRejected,
        icon: 'ic_notification',
        notificationIcon: 'ic_notification',
        color: rgbColor);
    ConnectycubeFlutterCallKit.onTokenRefreshed = (token) {
      debugPrint('CallEventModel:onTokenRefreshed: $token');
    };

    ConnectycubeFlutterCallKit.onCallAcceptedWhenTerminated = onCallAcceptedWhenTerminated;
    ConnectycubeFlutterCallKit.onCallRejectedWhenTerminated = onCallRejectedWhenTerminated;

...
}
```

**Comment by TatankaConCube** [comment]
@berger89  was the app started from the IDE or was it the application's regular start? Could you try to kill the app after starting it from the IDE and then start by clicking on the app icon from your launcher and reproduce this issue? Because the IDE starts the app using some running flags which prevent the calling of some callbacks from the killed state.

**Comment by TatankaConCube** [comment]
@devFan1 the annotation `@pragma('vm:entry-point')` for your handlers may help you to resolve your issue, please try to mark your handlers with this annotation

**Comment by berger89** [comment]
@TatankaConCube 

- started from the IDE: everything is working fine, `onCallAcceptedWhenTerminated` and `onCallRejectedWhenTerminated` are called when the app is terminated.

- started from the IDE -> kill the app -> start by clicking on the app icon: everything is working fine, `onCallAcceptedWhenTerminated` and `onCallRejectedWhenTerminated` are called when the app is terminated.

- build an APK -> install the app with the APK -> start the app: `onCallAcceptedWhenTerminated` and `onCallRejectedWhenTerminated` are NOT called when the app is terminated.

**Comment by TatankaConCube** [comment]
ok, thank you, guys, what version of the Flutter framework do you use in your projects?

**Comment by TatankaConCube** [comment]
@DooMMasteR @berger89 and others, do you use the [annotation](https://github.com/firebase/flutterfire/issues/9446#issuecomment-1240554285) as I recommended before? We just use a similar to the Firebase team logic of handling the callbacks and their recommendation applicable in this case.

**Comment by DooMMasteR** [comment]
See @berger89's code qute, the `@pragma` annotations are there...

**Comment by TatankaConCube** [comment]
could you provide the logcat output after clicking the Accept or Reject button?

**Comment by DooMMasteR** [comment]
Sry it is a bit dirty, I cleaned it up already:

```adb
-->> This is the Tap on the accep button:
04-04 12:24:00.432  1880  2244 D MIUIInput: [MotionEvent] publisher action=0x0, 567842768, channel '2593b0d com.ucware.mobile.app/com.connectycube.flutter.connectycube_flutter_call_kit.IncomingCallActivity (server)'
04-04 12:24:00.505  1880  2244 D MIUIInput: [MotionEvent] publisher action=0x1, 567842851, channel '2593b0d com.ucware.mobile.app/com.connectycube.flutter.connectycube_flutter_call_kit.IncomingCallActivity (server)'

04-04 12:24:00.529  1880  5236 I ActivityTaskManager: START u0 {act=action_call_accept cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=com.ucware.mobile.app cmp=com.ucware.mobile.app/.MainActivity} from uid 10646 from pid 25033 callingPackage com.ucware.mobile.app
04-04 12:24:00.530 23081 23081 I NotifySyncService_23081_23081: onNotificationRemoved appID:com.ucware.mobile.app uid:-987219659 key:0|com.ucware.mobile.app|-987219659|null|10646
04-04 12:24:00.532  1085  1198 E ANDR-PERF-LM: MetaMeter: handleAsync() 166: AdaptLaunch Invalid Async
04-04 12:24:00.532  1085  1198 E ANDR-PERF-LM: MetaMeter: handleAsync() 166: AdaptLaunch Invalid Async
04-04 12:24:00.533  1085  1198 E ANDR-PERF-LM: MetaMeter: handleAsync() 166: AdaptLaunch Invalid Async
04-04 12:24:00.533  1880  5236 E ANDR-PERF-JNI: com_qualcomm_qtiperformance_native_perf_io_prefetch_start
04-04 12:24:00.534 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.537  2869  3226 D PeopleSpaceWidgetMgr: Sbn doesn't contain valid PeopleTileKey: null/0/com.ucware.mobile.app
04-04 12:24:00.538 31897 31933 I Launcher.ApplicationsMessage: update com.ucware.mobile.app/ to null
04-04 12:24:00.541  1083  4159 E ANDR-IOP: io prefetch is disabled
04-04 12:24:00.541 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.541 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.542 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.542  1880  5236 D PerfImpl: perfWarmLaunchBoost: com.ucware.mobile.app, -1, 1
04-04 12:24:00.542 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.544 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.545 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.545 23457 23457 W DmNotificationServiceManager.ListenerService: sActiveNotifications.remove(com.ucware.mobile.app:-987219659) returned null for StatusBarNotification(pkg=com.ucware.mobile.app user=UserHandle{0} id=-987219659 tag=null key=0|com.ucware.mobile.app|-987219659|null|10646: Notification(channel=calls_channel_id shortcut=null contentView=null vibrate=null sound=null defaults=0x0 flags=0x92 color=0xff00b726 category=call actions=2 vis=PUBLIC))
04-04 12:24:00.551  1880  4321 D PerfImpl: perfColdLaunchBoost: com.ucware.mobile.app, 25033, 102
04-04 12:24:00.552  1085  1198 E ANDR-PERF-LM: MetaMeter: handleAsync() 166: AdaptLaunch Invalid Async
04-04 12:24:00.554  1880  6207 D PerfImpl: perfWarmLaunchBoost: com.ucware.mobile.app, -1, 1
04-04 12:24:00.554  1880  6207 D PerfImpl: perfAnimBoost: com.ucware.mobile.app
04-04 12:24:00.554  1880  6207 D PerfImpl: perfColdLaunchBoost: com.ucware.mobile.app, 25033, 102
04-04 12:24:00.554  1085  1198 E ANDR-PERF-LM: MetaMeter: handleAsync() 166: AdaptLaunch Invalid Async
04-04 12:24:00.566  1880  2327 D ProcessManager: skip notify foregroundAppInfo:ApplicationInfo{6888324 com.ucware.mobile.app}
04-04 12:24:00.601 31897 31897 D RecentsImpl: mActivityStateObserver com.ucware.mobile.app.MainActivity
04-04 12:24:00.601 31897 31897 W RecentsImpl: onResumed className=com.ucware.mobile.app.MainActivity   mIsInAnotherPro=false   isKeyguardLocked=true
04-04 12:24:00.615  1880  4321 D WindowManager: Input focus has changed to Window{8a4fd54 u0 com.ucware.mobile.app/com.ucware.mobile.app.MainActivity} display=0
04-04 12:24:00.627  1880  4321 D ActivityTaskManager:  setRequestedOrientationFreeForm: requestedOrientation= 1 task=Task{d1a25ea #60172 type=standard A=10646:com.ucware.mobile.app U=0 visible=true mode=fullscreen translucent=false sz=1} MiuiFreeformServicesUtils.getFreeformIgnoreRequestOrientationList()= [com.miui.player, com.sqw.setdl.mi, air.tv.douyu.android, com.youku.phone, com.tencent.gamehelper.smoba, com.taobao.litetao]packageName= com.ucware.mobile.app
04-04 12:24:00.629 25033 25033 D SurfaceView: UPDATE Surface(name=SurfaceView[com.ucware.mobile.app/com.ucware.mobile.app.MainActivity])/@0x87f2aeb, mIsProjectionMode = false
--> This is the error?!?
04-04 12:24:00.637 25033 25224 E flutter : [ERROR:flutter/shell/common/shell.cc(89)] Dart Error: Dart_LookupLibrary: library 'package:connectycube_flutter_call_kit/src/connectycube_flutter_call_kit.dart' not found.
04-04 12:24:00.637 25033 25224 E flutter : [ERROR:flutter/runtime/dart_isolate.cc(668)] Could not resolve main entrypoint function.
04-04 12:24:00.637 25033 25224 E flutter : [ERROR:flutter/runtime/dart_isolate.cc(168)] Could not run the run main Dart entrypoint.
04-04 12:24:00.637 25033 25224 E flutter : [ERROR:flutter/runtime/runtime_controller.cc(417)] Could not create root isolate.
04-04 12:24:00.637 25033 25224 E flutter : [ERROR:flutter/shell/common/shell.cc(606)] Could not launch engine with configuration.
04-04 12:24:00.654  1880  2612 W TelephonyPermissions: reportAccessDeniedToReadIdentifiers:com.ucware.mobile.app:getSerial:-1
04-04 12:24:00.659 31897 16408 W RecentsModel: getRunningTask   taskInfo=TaskInfo{userId=0 taskId=60172 displayId=0 isRunning=true baseIntent=Intent { act=action_call_accept cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=com.ucware.mobile.app cmp=com.ucware.mobile.app/.MainActivity } baseActivity=ComponentInfo{com.ucware.mobile.app/com.ucware.mobile.app.MainActivity} topActivity=ComponentInfo{com.ucware.mobile.app/com.ucware.mobile.app.MainActivity} origActivity=null realActivity=ComponentInfo{com.ucware.mobile.app/com.ucware.mobile.app.MainActivity} numActivities=1 lastActiveTime=612568435 supportsSplitScreenMultiWindow=true supportsMultiWindow=true resizeMode=1 isResizeable=true token=WCT{android.window.IWindowContainerToken$Stub$Proxy@452a0c0} topActivityType=1 pictureInPictureParams=PictureInPictureParams( aspectRatio=null sourceRectHint=null hasSetActions=false isAutoPipEnabled=false isSeamlessResizeEnabled=true) displayCutoutSafeInsets=Rect(0, 90 - 0, 0) topActivityInfo=ActivityInfo{8a921f9 com.ucware.mobile.app.MainActivity} launchCookies=[] positionInParent=Point(0, 0) parentTaskId=-1 isFocused=true isVisible=true topActivityInSizeCompat=false locusId= null windowMode=1}
04-04 12:24:00.659 31897 16408 W RecentsModel: getTaskInfoIgnoreHomeAndFreeform   taskInfo=ComponentInfo{com.ucware.mobile.app/com.ucware.mobile.app.MainActivity}
04-04 12:24:00.660 31897 16408 E ActivityManagerWrapper: get all recent tasks force including 60172
04-04 12:24:00.660 31897 16408 E ActivityManagerWrapper: getRecentTasks: size=3
04-04 12:24:00.660 31897 16408 E ActivityManagerWrapper: getRecentTasks: taskId=60172   userId=0   baseIntent=Intent { act=action_call_accept flag=268435456 cmp=ComponentInfo{com.ucware.mobile.app/com.ucware.mobile.app.MainActivity} }
04-04 12:24:00.660 31897 16408 E ActivityManagerWrapper: getRecentTasks: taskId=60171   userId=0   baseIntent=Intent { act=null flag=276824064 cmp=ComponentInfo{com.ucware.mobile.app/com.connectycube.flutter.connectycube_flutter_call_kit.IncomingCallActivity} }
04-04 12:24:00.660 31897 16408 E ActivityManagerWrapper: getRecentTasks: taskId=60167   userId=0   baseIntent=Intent { act=android.intent.action.MAIN flag=270532608 cmp=ComponentInfo{org.mozilla.firefox/org.mozilla.firefox.App} }
04-04 12:24:00.713  1880  1990 D PerfImpl: perfFirstDrawBoost: com.ucware.mobile.app, 25033, 1
04-04 12:24:00.713  1880  1986 I ActivityTaskManager: Displayed com.ucware.mobile.app/.MainActivity: +184ms
04-04 12:24:00.713  1880  1986 I ActivityTaskManager: Fully drawn com.ucware.mobile.app/.MainActivity: +184ms
04-04 12:24:00.716  1880  6207 W InputManager-JNI: Input channel object 'd6cb6b6 Splash Screen com.ucware.mobile.app (client)' was disposed without first being removed with the input manager!
```

**Comment by TatankaConCube** [comment]
> --> This is the error?!?

yes, but I still have not found how to resolve it, there is one suggestion, which is adding the annotation `@pragma('vm:entry-point')` but it doesn't work (((

**Comment by TatankaConCube** [comment]
I found a few ideas, will check them and back later.

**Comment by TatankaConCube** [comment]
@berger89 @DooMMasteR @devFan1 please try the new version [2.2.3](https://pub.dev/packages/connectycube_flutter_call_kit/versions/2.2.3), the issue should be fixed

**Comment by berger89** [comment]
The fix with version 2.2.3 worked for me. 

@TatankaConCube 
Thanks for the quick support!

**Comment by TatankaConCube** [comment]
closing cause some members reported about solving the issue
